pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -576,8 +618,10 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 		}
 		// End critical section.  It is safe to release the static IP (which
 		// just demotes it to ephemeral) now that it is attached.  In the case
-		// of a user-requested IP, the 'defer' won't be installed anyway.
-		releaseStaticIP = true
+		// of a user-requested IP, the "is user-owned" flag will be set,
+		// preventing it from actually being released.
+		isSafeToReleaseIP = true
```

Comment:
It really feels like we should be using a different variable for this part (the locking before/after deleting/creating the forwarding rule), to make sure we don't set this to true when up above we had set it to false